### DISCUSSION TITLE
Multiple relations between two tables?

### DISCUSSION THREAD

**Discussion by haratosan** [supabase/supabase#3239] [Questions] [answered]
I am currently developing a management tool for my sports school. Among other things, it will be used to plan courses.

Now I need the possibility to create multiple relations between two tables.

For example, I have a user table and one for the courses. In the courses the trainers are listed (can be several / linkd to the user table) as well as the participants (also from the user table). A separate trainer table is not an option, because they can also be participants in other lessons. Is there a possibility?

**Comment by dshukertjr** [comment, marked as answer]
Hi @budoteam!

You could simply create a relational table like this:

```sql
create table if not exists public.users (
  id uuid references auth.users on delete cascade not null primary key,
  name varchar(18) not null unique
);

create table if not exists public.courses (
  id uuid not null primary key DEFAULT uuid_generate_v4 (),
  title varchar(18) not null
);

create type participant_type AS ENUM ('trainer', 'participant');

create table if not exists public.participants (
  user_id uuid references public.users not null,
  course_id uuid references public.courses not null,
  type participant_type not null,
  primary key (user_id, course_id)
);
```

You have a `participants` table that holds whoever is participating in certain course. A participant can either be a `trainer` or a `partifipant` in a given course. The same user can be the other role in different courses.

**Reply by haratosan** [reply]
Hi @dshukertjr 

Thank you very much! 
With this solution, is it possible to fetch a lesson with the participants and trainers?

so the answer would look something like this?
`{
'id':1,
'day':'Monday'
'trainer':'[t1, t2]'
'participants':'[p1, p2]'
}`

I am trying to replace my existing backend with supabase and would be happy if I don't have to customize the whole frontend :-)

And what would an insert look like?

Thank you very much

**Reply by dshukertjr** [reply]
I could not get it exactly the way you wanted. Closest I could get was the following, so I hope it's good enough for your case. 

Maybe if you post a question regarding how to query the data as a separate question, one of the Supabase query wizards might be able to help you out. 

First, create a view that combines `users` table with `participants` table

```sql
create or replace view course_participants
as
    select
      public.users.id,
      public.users.name,
      -- other user information you have
      public.participants.course_id,
      public.participants.type
    from public.users
    join public.participants on public.participants.user_id = public.users.id;
```

With this view in place, you can run a query like this to get the list of courses along with the trainers and participants

```typescript
  // This will get all courses with participant and trainer information contained in them
  const { data, error } = await supabase
    .from('courses')
    .select(
      `
  *,
  trainers_and_participants:course_participants!participants_course_id_fkey(*)
  `
    )
```

The only thing about this query is that the form is not what you wanted. It will have participants and trainers in same array, so you will have to filter them accordingly on the client side. 

`{ 'id':1, 'day':'Monday' 'trainers_and_participants':'[p1, t1, p2]'  }`

I'm guessing that you might have like a `courses` page where you will have a list of courses there. In those courses page, you might want to show list of trainers, but not necessary list of participants (apologize if my assumption is incorrect). In that case, you will want to load all courses along with any trainers attending those courses. You could achieve that with the following query:


```typescript
  const { data, error } = await supabase
    .from('courses')
    .select(
      `
  *,
  trainers:course_participants!participants_course_id_fkey(*)
  `
    )
    .eq('trainers.type', 'trainer')
```

With this, you will not have any participant information in your query, only trainers.
